common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -248,19 +248,16 @@ def _hidden_content_student_view(self, context):
         display depending on whether staff is masquerading.
         """
         if not self._can_user_view_content():
-            subsection_format = (self.format or _("subsection")).lower()  # pylint: disable=no-member
+            date_reason = "course has ended" if getattr(self, 'self_paced', False) else "due date has passed"
 
-            # Translators: subsection_format refers to the assignment
-            # type of the subsection, such as Homework, Lab, Exam, etc.
             banner_text = _(
-                "Because the due date has passed, "
-                "this {subsection_format} is hidden from the learner."
-            ).format(subsection_format=subsection_format)
+                "Because the {date_reason}, this assignment is hidden from the learner."
+            ).format(date_reason=date_reason)
 
             hidden_content_html = self.system.render_template(
                 'hidden_content.html',
                 {
-                    'subsection_format': subsection_format,
+                    'self_paced': getattr(self, 'self_paced', False),
```

Comment:
I was surprised to see `self_paced` defined as an inheritable attribute on all xBlocks - and not just on the course_module.  

@rlucioni is that intentional?  Shouldn't `self_paced` be defined on just the course and not on each xBlock?  It's currently in both of these locations, which seems like an error:

- https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/course_module.py#L784
- https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/modulestore/inheritance.py#L214